core/chain_makers.go

```diff
@@ -205,6 +206,20 @@ func (b *BlockGen) AddUncle(h *types.Header) {
 	b.uncles = append(b.uncles, h)
 }
 
+// AddWithdrawal adds a withdrawal to the generated block.
+func (b *BlockGen) AddWithdrawal(w *types.Withdrawal) {
+	// The withdrawal will be assigned the next valid index.
+	var idx uint64
+	for i := b.i - 1; i >= 0; i-- {
+		if wd := b.chain[i].Withdrawals(); len(wd) != 0 {
```

Comment:
It can be wrong though. The chain maker can generate a batch of blocks from the given parent. 

Let's imagine a special case that all blocks generated have no withdrawal, while the parent does
have. In this case the withdrawal will be assigned a incorrect index(0).

Reply:
You are correct